src/Microsoft.AspNetCore.SignalR.Common/Protocol/HandshakeProtocol.cs

```diff
@@ -214,17 +215,17 @@ public static bool TryParseRequestMessage(ref ReadOnlySequence<byte> buffer, out
                                 completed = true;
                                 break;
                             default:
-                                throw new InvalidDataException($"Unexpected token '{reader.TokenType}' when reading handshake request JSON.");
+                                throw new InvalidDataException($"Unexpected token '{reader.TokenType}' when reading handshake request JSON. {GetPayloadAsString()}");
```

Comment:
Put something useful around the payload goop otherwise it looks totally out of place and apropos of nothing in the message.

Suggest: `Blah blah an error occurred. Message content: '{...}'`